core/src/utilities/helpers/usersetting-dialog-helpers.js

```diff
@@ -1,20 +1,56 @@
-import { GenericHelpers } from './generic-helpers';
+import { GenericHelpers, RoutingHelpers, IframeHelpers } from './';
 import { LuigiConfig } from '../../core-api';
 class UserSettingsHelperClass {
     constructor() { }
 
-    prepareUserSettingsSchema() {
-        const preparedSchema = [];
+
+    processUserSettingGroups() {
+        const userSettingGroups = [];
         const userSettingsSchema = LuigiConfig.getConfigValue('settings.userSettings.userSettingGroups');
         if (GenericHelpers.isObject(userSettingsSchema)) {
             for (const item in userSettingsSchema) {
                 let innerObj = {};
                 innerObj[item] = userSettingsSchema[item];
-                preparedSchema.push(innerObj);
+                userSettingGroups.push(innerObj);
             };
-            return preparedSchema;
+            return userSettingGroups;
+        }
+        return userSettingGroups;
+    }
+
+    createIframe(viewUrl, userSettingsGroup) {
+        const iframe = IframeHelpers.createIframe(
+            viewUrl,
+            undefined,
+            undefined,
+            'usersettings'
+        );
+        const iframeCtn = document.querySelector('.iframeUserSettingsCtn');
+        iframe.setAttribute('userSettingsGroup', userSettingsGroup);
+        iframe.userSettingsGroup = userSettingsGroup; //important for caching
+        iframeCtn.appendChild(iframe);
+        return iframe;
+    }
+
+    getUserSettingsIframesInDom() {
+        const iframeCtn = document.querySelector('.iframeUserSettingsCtn');
+        return iframeCtn ? iframeCtn.children : [];
+    }
+
+    hideUserSettingsIframe() {
+        this.getUserSettingsIframesInDom().forEach(iframe => {
+            iframe.style.display = 'none';
+        });
+    }
+
+    findActiveCustomUserSettingsIframe(eventSource){
+        let customUserSettingsIframes = document.querySelectorAll('[userSettingsGroup]');
+        for(let i = 0;i<customUserSettingsIframes.length;i++){
+            if(customUserSettingsIframes[i].contentWindow === eventSource){
+                return customUserSettingsIframes[i];
+            }
         }
```

Comment:
```suggestion
      return customUserSettingsIframes.find(iframe => iframe.contentWindow === eventSource);
```
if you add this suggestion, remember to remove line 53